sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala

```diff
@@ -344,10 +344,11 @@ case class DescribeRelation(
  */
 case class DescribeColumn(
     relation: LogicalPlan,
-    colNameParts: Seq[String],
+    column: NamedExpression,
     isExtended: Boolean) extends Command {
   override def children: Seq[LogicalPlan] = Seq(relation)
   override def output: Seq[Attribute] = DescribeCommandSchema.describeColumnAttributes()
+  override lazy val references: AttributeSet = AttributeSet.empty
```

Comment:
This is required since `column` is an expression, `QueryPlan.expressions` will pick up this field.

Reply:
Another way to approach is to use `LogicalPlan` instead of `NamedExpression`, but I went with this approach to reuse `UnresolvedAttribute`.

cc @cloud-fan

Reply:
What's the problem if `QueryPlan.expressions` pick up this field? Then we can let the analyzer to resolve columns which is even better.

Reply:
The issue is that in the following check, https://github.com/apache/spark/blob/303b8c87737fdff83c96855084c16d6504b0b50f/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala#L612-L618

`missingInput` (which is `references -- inputSet`) will be non-empty since `references` will be from `expressions` and `inputSet` will be empty since we are setting `ResolvedTable`'s output to `Nil`.

Reply:
One way to fix is to update `UnresolvedTable` as follow:
```scala
case class ResolvedTable(catalog: TableCatalog, identifier: Identifier, table: Table)
  extends LeafNode {
  override lazy val output: Seq[Attribute] = table.schema().toAttributes
}
```
Would you prefer this approach @cloud-fan ?

Reply:
One disadvantage of this approach is that since `ResolvedView` only stores the identifier, this doesn't work for views.

Reply:
If we need a hack anyway, I'd prefer we override `inputSet` here, so that we can let analyzer to resolve columns.

Reply:
This is one possible implementation of `inputSet` here:
```scala
  override lazy val inputSet: AttributeSet = {
    val output = relation match {
      case r: ResolvedTable => r.table.schema.toAttributes // creates new AttributeReference
      case _ => Seq() // should assert
    }
    AttributeSet(output)
  }
```
However, when `DescribeColumn` case class is copied in `ResolveAttribute` (`r.copy(column = resolvedCol)`), new `AttributeReference`s are created since `toAttributes` is called.

Any idea on the workaround (without updating `ResolvedTable` to have `output`)?

Reply:
If we want to use the existing `ResolveReferences`, we can override `resolveChildren` here, but it still requires maintaining the same attribute references. So I think introducing `output` at the `ResolvedTable` is the cleanest approach?

Reply:
I agree. About `ResolvedView`, I think eventually it will have output as well, after we have v2 view APII. For now, `ResolvedView` always go to v1 command, so we don't need its output.

Reply:
Updated.